lib/smart_proxy_remote_execution_ssh/runners/script_runner.rb

```diff
@@ -258,32 +261,38 @@ def indent_multiline(string)
     end
 
     def should_cleanup?
-      @session && !@session.closed? && @cleanup_working_dirs
+      @session && @cleanup_working_dirs
     end
 
-    def session
-      @session ||= begin
-                     @logger.debug("opening session to #{@ssh_user}@#{@host}")
-                     Net::SSH.start(@host, @ssh_user, ssh_options)
-                   end
+    def session(command)
+      @session = true
+
+      in_read, in_write = IO.pipe
+      out_read, out_write = IO.pipe
+      args = []
+      args += [{'SSHPASS' => @ssh_password}, '/usr/bin/sshpass', '-e'] if @ssh_password
+      args += ['/usr/bin/ssh', @host, ssh_options, command].flatten
+      command_pid = spawn(*args, :in => in_read, :out => out_write)
```

Comment:
Because `:err` is not redirected, ssh debug logging leaks to the parent process' stderr. This should be captured and passed to he job's output

Reply:
I am not sure if it is now captured properly, because when I used async ssh, stdout is sometimes mixed with stderr